test/core/rules/functional/test_segments.py

```diff
@@ -138,3 +139,25 @@ def test_segment_predicates_and():
         )
         == segments.Segments()
     )
+
+
+def test_segments_recursive_crawl():
+    """Test the "recursive_crawl()" function."""
+    sql = """
+    WITH cte AS (
+        SELECT * FROM tab_a
+    )
+    SELECT
+        cte.col_a,
+        tab_b.col_b
+    FROM cte
+    INNER JOIN tab_b;
+    """
+    linter = Linter()
+    parsed = linter.parse_string(sql)
+
+    functional_tree = segments.Segments(parsed.tree)
+
+    assert len(functional_tree.recursive_crawl("whitespace")) == 18
```

Comment:
Seems a bit of a flaky test (and annoying to calc/update if test changes). Would the two test cases below suffice?

Reply:
sounds good I just picked random segments

Reply:
updated now